### PR TITLE
[FLINK-10097][DataStream API] Additional tests for StreamingFileSink

### PR DESCRIPTION
**Pull request by kl0u**
## What is the purpose of the change

Adds more tests and partially refactors existing ones to decrease running time and increase test coverage. In addition, this improves also maintainability of the tests.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

R @aljoscha , @zentol

### CODE REVIEW

**Comment by kl0u** [comment]
R @aljoscha or @zentol ?

File: flink-streaming-java/src/test/java/org/apache/flink/streaming/api/functions/sink/filesystem/LocalStreamingFileSinkTest.java [hunk 1]
```diff
@@ -484,131 +479,4 @@ public void testScalingDownAndMergingOfStates() throws Exception {
 			Assert.assertEquals(3L, counter);
 		}
 	}
-
-	@Test
-	public void testMaxCounterUponRecovery() throws Exception {
```
Comment (azagrebin) [inline flink-streaming-java/src/test/java/org/apache/flink/streaming/api/functions/sink/filesystem/LocalStreamingFileSinkTest.java]: Is this test not relevant anymore?

File: flink-streaming-java/src/test/java/org/apache/flink/streaming/api/functions/sink/filesystem/RollingPolicyTest.java [hunk 2]
```diff
@@ -50,233 +43,234 @@
...
-			testHarness.setProcessingTime(0L);
+		final Buckets<String, String> buckets = createBuckets(path, rollingPolicy);
 
-			testHarness.processElement(new StreamRecord<>(Tuple2.of("test2", 1), 1L));
+		rollingPolicy.verifyCallCounters(0L, 0L, 0L, 0L, 0L, 0L);
 
-			testHarness.processElement(new StreamRecord<>(Tuple2.of("test1", 1), 1L));
-			testHarness.processElement(new StreamRecord<>(Tuple2.of("test1", 2), 2L));
-			TestUtils.checkLocalFs(outDir, 2, 0);
+		// the following 2 elements will close a part file because of size...
```
Comment (azagrebin) [inline flink-streaming-java/src/test/java/org/apache/flink/streaming/api/functions/sink/filesystem/RollingPolicyTest.java]: the comment is probably from the next test

**Comment by kl0u** [comment]
Thanks a lot  @azagrebin ! I will integrate your comments and merge.

File: flink-streaming-java/src/test/java/org/apache/flink/streaming/api/functions/sink/filesystem/LocalStreamingFileSinkTest.java (hunk 1 above)
Comment (kl0u) [reply flink-streaming-java/src/test/java/org/apache/flink/streaming/api/functions/sink/filesystem/LocalStreamingFileSinkTest.java]: It is relevant but it is subsumed by the `BucketsTest.testMergeAtScaleInAndMaxCounterAtRecovery()`.

File: flink-streaming-java/src/test/java/org/apache/flink/streaming/api/functions/sink/filesystem/RollingPolicyTest.java (hunk 2 above)
Comment (kl0u) [reply flink-streaming-java/src/test/java/org/apache/flink/streaming/api/functions/sink/filesystem/RollingPolicyTest.java]: You are right!

**Comment by kl0u** [comment]
I will wait for Travis to give green and then I will merge.

**Pull request closed without merging**